### PR TITLE
Ts 369 graphql documentation

### LINKED ISSUE #369: GPL-nnn As developers we would like documentation for the api v2 end points
**User story**
As developers we would like documentation for the api v2 end points

**Who the primary contacts are for this story**
Steve / Harry

**Acceptance criteria**
To be considered successful the solution must allow:
1. When I go to the api/v2 I should be able to see some usable documentation so that I can see how to use the endpoints without looking through the code.

**Additional context**

### PR DESCRIPTION
**Pull request by sdjm-tessella**
Closes #369

Changes proposed in this pull request:

* Add dependency on GraphQL Docs.
* Change the way we generate the GraphQL schema files.
* Create docs as a rake task from the most recent schema files.
* Add descriptions to all fields, types and arguments for GraphQL.

**NOTE:**  A lot of files here were auto-generated, so you can probably ignore everything in `.sass-cache` and `app/views/graphql`

### CODE REVIEW

File: app/graphql/types/outputs/tube_type.rb [hunk 1]
```diff
@@ -4,8 +4,8 @@ module Types
   module Outputs
     # The type for Tube objects.
     class TubeType < CommonOutputObject
-      field :barcode, String, null: false
-      field :materials, [MaterialUnionType], null: true
+      field :barcode, String, 'The barcode of this tube.', null: false
+      field :materials, [MaterialUnionType], 'The material contained in this tube.', null: true
```
Comment (Chris-Friend) [inline app/graphql/types/outputs/tube_type.rb:8]: Worth pluralising _material_ in the description?

File: app/graphql/types/outputs/well_type.rb [hunk 2]
```diff
@@ -4,9 +4,10 @@ module Types
   module Outputs
     # The type for Well objects.
     class WellType < CommonOutputObject
-      field :position, String, null: false
-      field :plate_id, Integer, null: false
-      field :materials, [MaterialUnionType], null: true
+      field :position, String, 'The description of the position for this well on the plate.',
+            null: false
+      field :plate_id, ID, 'The ID of the plate this well belongs to.', null: false
+      field :materials, [MaterialUnionType], 'The material contained within this well.', null: true
```
Comment (Chris-Friend) [inline app/graphql/types/outputs/well_type.rb:10]: Same here?

File: app/graphql/types/outputs/tube_type.rb (hunk 1 above)
Comment (sdjm-tessella) [reply app/graphql/types/outputs/tube_type.rb:8]: Oops yes

**Comment by sdjm-tessella** [comment]
> Can you add .sass-cache and app/views/graphql ot gitignore?

You don't want the documentation version controlled @stevieing ?

**Pull request merged**